src/components/Extension/Notification.js

```diff
@@ -0,0 +1,105 @@
+import { Component } from 'react'
+import { connect } from 'react-redux'
+
+import { notificationEmitted } from 'actions/notificationActions'
+import { fillTemplate } from 'services/playbookService'
+
+class Notification extends Component {
+  componentDidMount () {
+    if (!this.props.isEmitted) {
+      this.createNotification()
+    }
+  }
+
+  render () {
+    if (this.props.isClicked) {
+      this.handleClicks()
+    }
+
+    return null
+  }
+
+  handleClicks () {
+    if (!chrome || !chrome.tabs) {
+      return
+    }
+
+    const { buttonActions, buttonIndex } = this.props
+
+    const url = buttonIndex
+              ? buttonActions[buttonIndex].url
+              : buttonActions.find(buttonAction => buttonAction.default).url
+
+    chrome.tabs.create({ url })
+  }
+
+  createNotification () {
+    const {
+      eventData,
+      eventId,
+      incidentId,
+      notificationEmitted,
+      actions,
+      event,
+      ticket,
+      eventType,
+      engagement
+    } = this.props
+
+    const ticketId = ticket.originId
+    const status = ticket.status
+    const severity = ticket.data ? ticket.data.severity : null
+    const title = ticket.data ? ticket.data.title : null
+
+    const message = eventData.DisplayText ||
+                    Object.entries(eventData)
+                          .map(([key, value]) => `${key}: ${value}`)
+                          .join('\n')
+
+    const defaultActions = [{
+      title: 'Open in IcM', // TODO: Ticket system hardcoded
+      url: `https://icm.ad.msft.net/imp/v3/incidents/details/${ticketId}/home`
+    }, {
+      title: 'Open in SIA',
+      url: `https://sia-dev-eventui.azurewebsites.net/incident/${incidentId}`,
+      default: true
+    }]
+
+    const playbookActions = actions
+      ? actions.filter(action => action.actionTemplate.isUrl)
+               .map(action => ({
+                 title: `${action.name}: ${action.actionTemplate.name}`,
+                 url: fillTemplate(action.actionTemplate, event, ticket, eventType, engagement)
+               })) : []
+
+    const buttonActions = playbookActions.concat(defaultActions)
+
+    const notificationBody = {
+      type: 'basic',
+      iconUrl: `/static/sia-blue-icon-128.png`,
+      title: `${ticketId} | ${status} ${severity ? '| Sev ' + severity : ''}`,
+      message: message,
+      contextMessage: title || `SRE Incident Assistant | Event ID: ${eventId}`,
+      // Only two buttons allowed in chrome notifications
+      buttons: buttonActions.map(buttonAction => ({
+        title: buttonAction.title,
+        iconUrl: '/static/sia-green-icon-128.png'
+      }))
+    }
+
+    const msg = `############## EventId: ${this.props.eventId} -- createNotification.`
+    console.log('%c' + msg, 'color:Green;font-weight:bold;', this.props, notificationBody)
+
+    if (chrome && chrome.notifications) {
```

Comment:
Is there even a reason to render mount this component if this condition isn't met?

Reply:
You are right. This was for debugging without having to install the extension. Will make the change.